### PR TITLE
[CP-756] Backup password confirmation

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-756]

**Description**
Implement password confirmation functionality for backups

<details>
<summary><b>Screenshots</b></summary>
// put images here
Screen Shot 2021-11-10 at 3 46 12 PM
</details>

[CP-756]: https://appnroll.atlassian.net/browse/CP-756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx [hunk 1]
```diff
@@ -43,6 +44,12 @@ const messages = defineMessages({
   backupSetSecretKeyModalInputLabel: {
     id: "module.overview.backupSetSecretKeyModalInputLabel",
   },
+  backupSetConfirmationSecretKeyModalInputLabel: {
+    id: "module.overview.backupSetConfirmationSecretKeyModalInputLabel",
+  },
+  backupPasswordConfirmationDoesntMatch: {
```
Comment (dkarski) [inline packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx:50]: 🟡

```suggestion
  backupSecretKeyConfirmationDoesntMatch: {
```

`SecretKey` or `Password`. The password is a text so I understand the intention. However, I think that we should single semantic in a base code

File: packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx [hunk 2]
```diff
@@ -118,14 +125,27 @@ export const BackupSetSecretKeyModal: FunctionComponent<BackupSetSecretKeyModalP
           message={messages.backupSetSecretKeyModalDescription}
         />
         <Form onSubmit={handleSubmitClick}>
```
Comment (dkarski) [inline packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx:127]: 🟢
> You shouldn't be able to continue without entering your password

[source](https://appnroll.slack.com/archives/C020D6DUC1W/p1636464523019800?thread_ts=1636462775.015400&cid=C020D6DUC1W)

Could you please handle this requirement as part of this PR? 🙏 It's my bad that I didn't update the story after an Aleksandra Rogucka comment -> I'm sorry 😞

File: packages/app/src/ui/components/password-field/password-filed.component.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+interface PasswordInputProps {
+  errorMessage?: string
+  showPassword?: boolean
+  label: {
+    id: string
+  }
+}
+
+export const PasswordInput: FunctionComponent<PasswordInputProps> = ({
```
Comment (dkarski) [inline packages/app/src/ui/components/password-field/password-filed.component.tsx:27]: 🟡
```suggestion
export const PasswordField: FunctionComponent<PasswordInputProps> = ({
```
`PasswordField` or `PasswordInput`

File: packages/app/src/ui/components/password-field/password-filed.component.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import Icon, { IconSize } from "Renderer/components/core/icon/icon.component"
+import { PasswordFieldTestIds } from "App/ui/components/password-field/password-field-test-ids.enum"
+import {
+  FieldWrapper,
+  FormInputLabel,
+  FormInput,
+  IconWrapper,
+} from "App/ui/components/password-field/password-field.styled"
+
+interface PasswordInputProps {
```
Comment (dkarski) [inline packages/app/src/ui/components/password-field/password-filed.component.tsx:19]:
```suggestion
interface Props {
```

**Comment by dkarski** [comment]
Great idea with `App/ui` intention 👏 👌

File: packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx]: https://github.com/mudita/mudita-center/pull/518/commits/7a1d7d36113bcc0010d3fa2fd8708f496309465e

File: packages/app/src/ui/components/password-field/password-filed.component.tsx (hunk 3 above)
Comment (igorbezsmertnyi) [reply packages/app/src/ui/components/password-field/password-filed.component.tsx]: https://github.com/mudita/mudita-center/pull/518/commits/7a1d7d36113bcc0010d3fa2fd8708f496309465e

File: packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx (hunk 2 above)
Comment (igorbezsmertnyi) [reply packages/app/src/overview/components/backup-set-secret-key-modal-dialog/backup-set-secret-key-modal-dialog.component.tsx:130]: https://github.com/mudita/mudita-center/pull/518/commits/7c4332fa0bf00026c792a0ff5d8dd2ef3eed2aa5

**Review by dkarski — APPROVED** [review]

**Review by wercix — APPROVED** [review]

**Pull request merged**